ROADMAP.md

```diff
@@ -1,177 +1,94 @@
-# Kubeflow Pipelines Roadmap
+### KFP ROADMAP
 
-## Kubeflow Pipelines 2022 Roadmap
+The purpose of this KFP Roadmap document is to outline the planned themes and items for future development and improvements of Kubeflow Pipelines (KFP). It categorizes these efforts into several key areas: User Experience, User Documentation, Core Platform Stability, Extensibility, Integrations, and Quality of Life, QE, Performance & Scalability, Security & Compliance, and Community & Governance. The document also provides information on how to contribute to KFP.
 
-### KFP v2
+## New Features & Enhancements
 
-Design: [bit.ly/kfp-v2](https://bit.ly/kfp-v2)
+Q4 2025
+- [ ] **Reusable components for common GenAI use cases**: As a GenAI practitioner, I want pre-built KFP components for common tasks like RAG and LLM compression to accelerate development.
+- [ ] **Workspace Support**: As a user, I want to share data across components within a pipeline. \[[KEP](https://github.com/kubeflow/pipelines/blob/master/proposals/11875-pipeline-workspace/README.md)\]
+- [ ] **KFP Local Mode Parity with Remote KFP**: As a user, I want the same feature set available to me in KFP local mode as KFP remote. \[[Epic](https://github.com/kubeflow/pipelines/issues/11793)\]
+- [ ] **Support for External Metadata Providers**: Allow KFP to support logging metadata to external providers such as MLFlow, WandB, etc..
+- [ ] **Multi-User Standalone KFP Deployment**: As a KFP admin, I want to deploy KFP standalone in multi-user mode for both production and development.
 
-#### KFP v2 Goals
+Q1 2026
+- [ ] **PVC as the Storage Backend (Alternative to S3)**: As a K8s admin, I would like to utilize K8s PVC and storage class abstractions for artifact storage.
+- [ ] **Share Python Code between Components in a Pipeline Version**
+- [ ] **KFP Events Support**: As a user, I want to trigger Pipeline runs, and potentially components based on events.
 
-* Enable v2 authoring Critical User Journeys
-  * Containerized Python component
-  * Bring-your-own container
-  * DAG component
-* Finalize IR and component packaging format to address feature gaps with v1
+Q2 2026
+- [ ] **100% Kubernetes Infra Abstraction in SDK**: As a data scientist, I don’t want to see Kubernetes in my SDK code.
+- [ ] **KFP UI with KFP Local Mode**: As a user, I want to be able to spin up a KFP UI server which runs KFP locally, allowing for a low resource intensive and quick mode of KFP deployment locally.
+- [ ] **Allow Deleting & Cleaning up Artifacts**
+- [ ] **Multi-Cluster Support**
```

Comment:
Is there a usecase that would warrant this? This seems very high effort depending on what you mean by multi-cluster.

Reply:
We've seen in the past users (specifically IBM users in this case) try to do their own multi-cluster KFP setups, here's a link to their slides they presented last year: https://drive.google.com/file/d/1vjg_5nSFraxjxoAwIV9VU8bg9uraMbCh/view - and more recently the engagements with granite build, my understanding was they also want support for pipeline runs that can execute in remote cluster. 

My thinking was it wouldn't be too difficult to enable since KFP is pretty stateless, and launchers/drivers can just connect to kfp server remotely, if you have an argo workflow controller already installed in your remote clusters. It might be more tricky with the kubernetes native deployment model though. 

I'm not attached to this, we can take it out until we see more requests for this.